### ISSUE TITLE
Failure with zero byte file

### ISSUE THREAD

**Issue by mapsam** [mapbox/mapbox-file-sniff#59]
The following error is reported when a zero byte file is run through mapbox-file-sniff:

```
fs.js:582
  binding.read(fd, buffer, offset, length, position, req);
          ^

Error: Offset is out of bounds
    at Error (native)
    at Object.fs.read (fs.js:582:11)
    at /Users/mapsam/mapbox/mapbox-file-sniff/index.js:34:16
    at FSReqWrap.oncomplete (fs.js:82:15)
```

We should add a `stats.size` check before reading a file to avoid this (even though fs.read is async, we hard fail on this and it doesn't return the callback).

cc @millzpaugh @GretaCB
